src/util/union_find.cpp

```diff
@@ -80,13 +80,13 @@ void unsigned_union_find::re_root(size_type old_root, size_type new_root)
   old_root=find(old_root);
 
   // same set?
-  // assert(find(new_root)==old_root);
+  // assert(find(new_root)==old_root); // Commented out dead code? To review
   if(find(new_root)!=old_root)
     return;
 
   // make sure we actually do something
-  assert(new_root!=old_root);
-  assert(nodes[old_root].count>=2);
+  INVARIANT(new_root!=old_root, "Ensure the algorithm roots differ"); // TODO review
+  INVARIANT(nodes[old_root].count>=2, ""); // TODO review suitable statement
```

Comment:
This actually sounds like a precondition? It sure isn't established by any of the previous code in the function.